ocall/ta/ocall_ta.c

```diff
@@ -0,0 +1,505 @@
+/*
+ * Copyright (c) 2020, Microsoft Corporation
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+#include <string.h>
+
+#include <ocall_ta.h>
+
+#include <pta_ocall.h>
+#include <util.h>
+
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+void TA_DestroyEntryPoint(void)
+{
+}
+
+TEE_Result TA_OpenSessionEntryPoint(uint32_t param_types, TEE_Param params[4],
+			    void **sess_ctx)
+{
+	(void)param_types;
+	(void)params;
+	(void)sess_ctx;
+
+	return TEE_SUCCESS;
+}
+
+void TA_CloseSessionEntryPoint(void *sess_ctx)
+{
+	(void)sess_ctx;
+}
+
+/* ECALL has no parameters, OCALL has no parameters */
+static TEE_Result test1(uint32_t param_types,
+		        TEE_Param __unused params[TEE_NUM_PARAMS])
+{
+	TEE_Result res;
+	uint32_t eorig;
+
+	const uint32_t expected_pt = TEE_PARAM_TYPES(
+		TEE_PARAM_TYPE_NONE,
+		TEE_PARAM_TYPE_NONE,
+		TEE_PARAM_TYPE_NONE,
+		TEE_PARAM_TYPE_NONE
+	);
+
+	if (param_types != expected_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	res = TEE_InvokeCACommand(TEE_TIMEOUT_INFINITE, TA_OCALL_CA_CMD_TEST_1,
```

Comment:
minor: maybe prefer indent to `(`. (@jforissier, do we try to be strict on that?)